stellargraph/data/edge_splitter.py

```diff
@@ -98,8 +98,10 @@ def _train_test_split_homogeneous(
         """
         # minedges are those edges that if removed we might end up with a disconnected graph after the positive edges
         # have been sampled.
-        if keep_connected:
+
+        if (not self.g.is_directed()) and keep_connected:
```

Comment:
Why do we ignore `keep_connected` for directed graphs?

Reply:
unfortunately `nx.minimum_spanning_edges` doesn't support directed graphs.

Reply:
Found the equalivent algo for directed graphs! I've added the nx Edmond's algo for directed graphs in `EdgeSplitter`